src/problem/generator/fractionalEquation.ts

```diff
@@ -0,0 +1,306 @@
+import { ProblemGenerator } from "../ProblemGeneratorSpi";
+import { randomInt } from "../util/randomizer";
+import { calculategcd } from "../util/commonDivisor";
+import { SortByAlpha } from "@mui/icons-material";
+
+
+interface Params {
+    [index: number]: any[];
```

Comment:
we should avoid using any (to benefit from type safety)

Reply:
What kind of Datatyp would be better here?

Reply:
Maybe something like this:

```
type fractionalTerm = {
    b: number;
    c: number;
    d: number;
}

type Params = {
    terms: fractionalTerm[];
    solutions: number[];
    pols: number[];
}
```

I think type would be more suitable than interface here.
In order to get the count of solutions you can always do: `params.solutions.length`

Not sure what "pols" is ;-)